crates/ruff_python_ast/src/visitor/preorder.rs

```diff
@@ -7,64 +8,89 @@ use crate::{
 
 /// Visitor that traverses all nodes recursively in pre-order.
 pub trait PreorderVisitor<'a> {
+    #[inline(always)]
```

Comment:
nit: just `#[inline]`, even though I'd be surprised if that wasn't already inlined